File: LENTABP10/src/main/java/com/lenta/lentabp10/MobileObjectsService/models/MB_S_30.java
package com.lenta.lentabp10.MobileObjectsService.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;

// Модель справочника товаров
public class MB_S_30 {

    /// Номер товара
    @JsonProperty("MATERIAL")
    @Getter private String material;

    /// Длинное наименование
    @JsonProperty("NAME")
    @Getter private String name;

    /// Вид товара
    @JsonProperty("MATYPE")
    @Getter private String matype;

    /// Базисная единица измерения
    @JsonProperty("BUOM")
    @Getter private String buom;

    @JsonProperty("MATR_TYPE")
    @Getter private String matrType;

    /// Номер секции товара
    @JsonProperty("ABTNR")
    @Getter private String abtnr;

    /// Признак возвратности товара
    @JsonProperty("IS_RETURN")
    @Getter private String isReturn;

    /// Флаг "Акцизный товар"
    @JsonProperty("IS_EXC")
    @Getter private String exc;

    /// Флаг "Алкоголь"
    @JsonProperty("IS_ALCO")
    @Getter private String alco;

    @JsonProperty("MATKL")
    @Getter private String matkl;

    @JsonProperty("EKGRP")
    @Getter private String ekgrp;

    public MB_S_30(String material, String name, String matype, String buom, String matrType, String abtnr, String isReturn, String exc, String alco, String matkl, String ekgrp) {
        this.material = material;
        this.name = name;
        this.matype = matype;
        this.buom = buom;
        this.matrType = matrType;
        this.abtnr = abtnr;
        this.isReturn = isReturn;
        this.exc = exc;
        this.alco = alco;
        this.matkl = matkl;
        this.ekgrp = ekgrp;
    }

    public boolean isAlco()
    {
        return alco.equals("X");
    }

    public boolean IsExcise()
    {
        return exc.equals("X");
    }
}

File: writeoff_breakage/src/main/java/com/lenta/bp10/mobileObjectsService/models/MB_S_30.kt
package com.lenta.bp10.mobileObjectsService.models

import com.google.gson.annotations.SerializedName

// Модель справочника товаров
class MB_S_30(
        /// Номер товара
        @SerializedName("MATERIAL")
        val material: String,

        /// Длинное наименование
        @SerializedName("NAME")
        val name: String,

        /// Вид товара
        @SerializedName("MATYPE")
        val matype: String,

        /// Базисная единица измерения
        @SerializedName("BUOM")
        val buom: String,

        @SerializedName("MATR_TYPE")
        val matrType: String,

        /// Номер секции товара
        @SerializedName("ABTNR")
        val abtnr: String,

        /// Признак возвратности товара
        @SerializedName("IS_RETURN")
        val isReturn: String,

        /// Флаг "Акцизный товар"
        @SerializedName("IS_EXC")
        val exc: String,

        /// Флаг "Алкоголь"
        @SerializedName("IS_ALCO")
        val alco: String,

        @SerializedName("MATKL")
        val matkl: String?,

        @SerializedName("EKGRP")
        val ekgrp: String?
) {
    fun isAlco(): Boolean {
        return alco == "X"
    }

    fun IsExcise(): Boolean {
        return exc == "X"
    }
}